src/walk-manifest.js

```diff
@@ -70,30 +71,31 @@ var parseKey = function(basedir, decrypt, resources, manifest, parent) {
 		));
 		key.uri = keyUri;
 		resources.push(key);
-		return key;
+		return callback(key);
 	}
 
 	// get the aes key
-	var keyContent = syncRequest('GET', keyUri).getBody();
-	key.bytes = new Uint32Array([
-		keyContent.readUInt32BE(0),
-		keyContent.readUInt32BE(4),
-		keyContent.readUInt32BE(8),
-		keyContent.readUInt32BE(12)
-	]);
-
-	// remove the key from the manifest
-	manifest.content = new Buffer(manifest.content.toString().replace(
-		new RegExp('.*' + key.uri + '.*'),
-		''
-	));
-
-
-	return key;
+  request(keyUri, function (error, response, body) {
+    var keyContent = body;
+    key.bytes = new Uint32Array([
+      keyContent.readUInt32BE(0),
+      keyContent.readUInt32BE(4),
+      keyContent.readUInt32BE(8),
+      keyContent.readUInt32BE(12)
+    ]);
+
+    // remove the key from the manifest
+    manifest.content = new Buffer(manifest.content.toString().replace(
+      new RegExp('.*' + key.uri + '.*'),
+      ''
+    ));
+
+
+    return callback(key);
+  });
 };
 
-var walkPlaylist = function(decrypt, basedir, uri, parent, manifestIndex) {
-	var resources = [];
+var walkPlaylist = function(decrypt, basedir, uri, parent, manifestIndex, resources, callback) {
```

Comment:
echoing Gary's comment above: resource accumulation should preferably be moved into the callback rather than using an out parameter.